### ISSUE TITLE
Scaffold DashboardMainApp and DashboardEntityApp

### ISSUE THREAD

**Issue by tofumatt** [google/site-kit-wp#4030]
## Feature Description

The [`GoogleSiteKitDashboard` component](https://github.com/google/site-kit-wp/blob/01faf4805f14baa86d60314106b955c71a0776c1/assets/js/googlesitekit-dashboard.js#L37) is the entry-point to the dashboard and can be set to render differently based on the [`unifiedDashboard ` feature flag](https://github.com/google/site-kit-wp/issues/4029). When the feature flag for `unifiedDashboard` is disabled, the existing `DashboardApp` component should render. When the feature flag for `unifiedDashboard` is enabled, we should instead render either `DashboardMainApp` or `DashboardEntityApp` based on the presence of the `permaLink` query param.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Two new components should be scaffolded in the `assets/js/components` folder: `DashboardMainApp` and `DashboardEntityApp`. They should only render the Site Kit Header (`assets/js/components/Header.js`) component for now.
* The [`GoogleSiteKitDashboard` component](https://github.com/google/site-kit-wp/blob/01faf4805f14baa86d60314106b955c71a0776c1/assets/js/googlesitekit-dashboard.js#L37) should, when `unifiedDashboard` feature flag is enabled: render `DashboardMainApp` when there is no `permaLink` query param and render `DashboardEntityApp` when the `permaLink` query param exists.
* If the `unifiedDashboard` feature flag is not enabled: `GoogleSiteKitDashboard` should behave as it currently does today.

## Implementation Brief

* 

### Test Coverage

* 

### Visual Regression Changes

* 

## QA Brief

* 

## Changelog entry

*

**Comment by aaemnnosttv** [comment]
IB ✅

**Comment by johnPhillips** [comment]
> The GoogleSiteKitDashboard component should, when unifiedDashboard feature flag is enabled: render DashboardMainApp when there is no permaLink query param and render DashboardEntityApp when the permaLink query param exists.

@aaemnnosttv I think we might have missed a step, looking at this more closely. 

It seems [if the `permaLink` query var is present, the PHP layer loads an entirely separate div](https://github.com/google/site-kit-wp/blob/861db7225e67b969e69a3afd7f851016096a7859/includes/Core/Admin/Screens.php#L295): `googlesitekit-dashboard-details`, into which the `DashboardDetailsApp` is loaded, so it never even hits 

What's the best way to proceed here? It feels a bit funny to test for the feature flag in the PHP layer, so I guess we need to add an override inside `assets/js/googlesitekit-dashboard-details.js` (or possibly `assets/js/components/dashboard-details/DashboardDetailsApp.js`) that returns the `DashboardEntityApp` if the feature flag is on?

Let me know what you think and I'll update the IB 👍

**Comment by aaemnnosttv** [comment]
@johnPhillips the `DashboardEntityApp` is essentially the unified dashboard version of the `DashboardDetailsApp` so we would conditionally render that one based on the feature flag, just as we would do with the current dashboard and unified "main" version of that. So no changes needed on the PHP side for now I don't think 🙂

**Comment by johnPhillips** [comment]
Thanks @aaemnnosttv. IB updated, let me know what you think?

**Comment by aaemnnosttv** [comment]
> Inside `assets/js/googlesitekit-dashboard.js`
> * If the `permalink` query argument is present, render the new  `DashboardEntityApp`  component

@johnPhillips this entrypoint would remain specific to the main dashboard. I see now that this is how it is defined in the ACs which I think may be a mistake. As far as I understand it, `GoogleSitekitDashboard` only decides to render `DashboardApp` or the new `DashboardMainApp` based on the feature flag. Similarly, the `assets/js/googlesitekit-dashboard-details.js` entry would conditionally render `DashboardDetailsApp` or the new `DashboardEntityApp` as you have defined – no query param checks should be needed in either case because that is still done on page load and controls which entry and render target are loaded.

@tofumatt can you confirm and if so, would you please update the ACs?

**Comment by felixarntz** [comment]
@aaemnnosttv @tofumatt Yes I see that now too. While we eventually want to combine these things into one app, let's keep things separate as is for now, in their own JS assets - this is related to the conversations we've had before about full page reloads between the two.

We can change this in the future, as 1-2 issues outside of this epic.

**Comment by tofumatt** [comment]
Indeed, this was an oversight in the ACs and design doc, because I actually just assumed the rendering logic was located clientside.

I've updated the ACs and have sent it back to you for IB @johnPhillips. Sorry about that! 😓

**Comment by johnPhillips** [comment]
No problem @tofumatt. This all makes sense to me now and I've updated the IB to remove references to the query param check.

**Comment by eugene-manuilov** [comment]
IB :heavy_check_mark:
